File: src/main/java/com/castgroup/service/PersonService.java
package com.castgroup.service;

import java.util.List;

import com.castgroup.dto.PersonDto;

/**
 * PersonService contains all the methods that needs to be implemented to manipulate Person informations.
 */
public interface PersonService {
	List<PersonDto> getAllPersons();
	PersonDto getPersonById(Integer id);
    void savePerson(PersonDto personDto);
    void deletePersonById(Integer id);    
}

File: src/main/resources/static/js/app/PersonService.js
angular.module('person.service', []).factory('PersonService',
		[ "$http", "CONSTANTS", function($http, CONSTANTS) {
			var service = {};
			service.getPersonById = function(id) {
				return $http.get(CONSTANTS.getPersonById + id);
			}
			service.getAllPersons = function() {
				return $http.get(CONSTANTS.getAllPersons);
			}
			service.savePerson = function(personDto) {
				return $http.post(CONSTANTS.savePerson, personDto);
			}
			service.deletePersonById = function(id){
				return $http.delete(CONSTANTS.deletePersonById + id);
			}
			return service;
		} ]);